In [ ]:
Arkan Putra Ridhana

Sumber Data:  
https://www.kaggle.com/datasets/atharvasoundankar/global-water-consumption-dataset-2000-2024  
Lisensi:  
No Copyright  
Author: -  
Collaborator: Atharva Soundankar  
Waktu Pengumpulan Data: 2000-2024  
Bagaimana Pengumpulan Data: Tidak tertera  

Penjelasan Data:
| Kolom | Arti | Jenis Variabel | Skala | Satuan | Contoh |
| :--- | :--- | :--- | :--- | :--- | :--- |
| `Country` | Nama negara tempat pengukuran diambil | Kualitatif | Nominal | – | Argentina |
| `Year` | Tahun pencatatan data (2000–2024) | Kuantitatif diskrit | Interval | Tahun | 2000 |
| `Total Water Consumption (Billion Cubic Meters)` | Total konsumsi air bersih tahunan suatu negara | Kuantitatif kontinu | Rasio | Miliar m³ | 481.49 |
| `Per Capita Water Use (Liters per Day)` | Rata-rata konsumsi air bersih harian per orang | Kuantitatif kontinu | Rasio | Liter/hari | 235.43 |
| `Agricultural Water Use (%)` | Persentase penggunaan air di sektor pertanian | Kuantitatif kontinu | Rasio | % | 48.55 |
| `Industrial Water Use (%)` | Persentase penggunaan air di sektor industri | Kuantitatif kontinu | Rasio | % | 20.84 |
| `Household Water Use (%)` | Persentase penggunaan air di sektor rumah tangga | Kuantitatif kontinu | Rasio | % | 30.10 |
| `Rainfall Impact (Annual Precipitation in mm)` | Curah hujan tahunan yang diterima negara tersebut | Kuantitatif kontinu | Rasio | mm | 1288.70 |
| `Groundwater Depletion Rate (%)` | Tingkat penurunan/penyusutan cadangan air tanah tahunan | Kuantitatif kontinu | Rasio | % | 3.26 |
| `Water Scarcity Level` | Tingkat keparahan kelangkaan air bersih | Kualitatif | Ordinal | – | Moderate |

Sumber Data:  
https://www.kaggle.com/datasets/atharvasoundankar/global-water-consumption-dataset-2000-2024  
Lisensi:  
No Copyright  
Author: -  
Collaborator: Atharva Soundankar  
Waktu Pengumpulan Data: 2000-2024  
Bagaimana Pengumpulan Data: Tidak tertera  

Penjelasan Data:
| Kolom | Arti | Jenis Variabel | Skala | Satuan | Contoh |
| :--- | :--- | :--- | :--- | :--- | :--- |
| `Country` | Nama negara tempat pengukuran diambil | Kualitatif | Nominal | – | Argentina |
| `Year` | Tahun pencatatan data (2000–2024) | Kuantitatif diskrit | Interval | Tahun | 2000 |
| `Total Water Consumption (Billion Cubic Meters)` | Total konsumsi air bersih tahunan suatu negara | Kuantitatif kontinu | Rasio | Miliar m³ | 481.49 |
| `Per Capita Water Use (Liters per Day)` | Rata-rata konsumsi air bersih harian per orang | Kuantitatif kontinu | Rasio | Liter/hari | 235.43 |
| `Agricultural Water Use (%)` | Persentase penggunaan air di sektor pertanian | Kuantitatif kontinu | Rasio | % | 48.55 |
| `Industrial Water Use (%)` | Persentase penggunaan air di sektor industri | Kuantitatif kontinu | Rasio | % | 20.84 |
| `Household Water Use (%)` | Persentase penggunaan air di sektor rumah tangga | Kuantitatif kontinu | Rasio | % | 30.10 |
| `Rainfall Impact (Annual Precipitation in mm)` | Curah hujan tahunan yang diterima negara tersebut | Kuantitatif kontinu | Rasio | mm | 1288.70 |
| `Groundwater Depletion Rate (%)` | Tingkat penurunan/penyusutan cadangan air tanah tahunan | Kuantitatif kontinu | Rasio | % | 3.26 |
| `Water Scarcity Level` | Tingkat keparahan kelangkaan air bersih | Kualitatif | Ordinal | – | Moderate |

Sumber Data:  
https://www.kaggle.com/datasets/atharvasoundankar/global-water-consumption-dataset-2000-2024  
Lisensi:  
No Copyright  
Author: -  
Collaborator: Atharva Soundankar  
Waktu Pengumpulan Data: 2000-2024  
Bagaimana Pengumpulan Data: Tidak tertera  

Penjelasan Data:
| Kolom | Arti | Jenis Variabel | Skala | Satuan | Contoh |
| :--- | :--- | :--- | :--- | :--- | :--- |
| `Country` | Nama negara tempat pengukuran diambil | Kualitatif | Nominal | – | Argentina |
| `Year` | Tahun pencatatan data (2000–2024) | Kuantitatif diskrit | Interval | Tahun | 2000 |
| `Total Water Consumption (Billion Cubic Meters)` | Total konsumsi air bersih tahunan suatu negara | Kuantitatif kontinu | Rasio | Miliar m³ | 481.49 |
| `Per Capita Water Use (Liters per Day)` | Rata-rata konsumsi air bersih harian per orang | Kuantitatif kontinu | Rasio | Liter/hari | 235.43 |
| `Agricultural Water Use (%)` | Persentase penggunaan air di sektor pertanian | Kuantitatif kontinu | Rasio | % | 48.55 |
| `Industrial Water Use (%)` | Persentase penggunaan air di sektor industri | Kuantitatif kontinu | Rasio | % | 20.84 |
| `Household Water Use (%)` | Persentase penggunaan air di sektor rumah tangga | Kuantitatif kontinu | Rasio | % | 30.10 |
| `Rainfall Impact (Annual Precipitation in mm)` | Curah hujan tahunan yang diterima negara tersebut | Kuantitatif kontinu | Rasio | mm | 1288.70 |
| `Groundwater Depletion Rate (%)` | Tingkat penurunan/penyusutan cadangan air tanah tahunan | Kuantitatif kontinu | Rasio | % | 3.26 |
| `Water Scarcity Level` | Tingkat keparahan kelangkaan air bersih | Kualitatif | Ordinal | – | Moderate |

In [9]:
import pandas as pd
df = pd.read_csv("dataset.csv")

print("Informasi tipe data: ")
df.info()

print("\nJumlah data kosong per kolom: ")
print(df.isna().sum())

print("\nJumlah baris duplikat: ")
print(df.duplicated().sum())

print("\n Ringkasan nilai min max: ")
print(df.describe().T[['min', 'max']])

Informasi tipe data: 
<class 'pandas.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 10 columns):
 #   Column                                          Non-Null Count  Dtype  
---  ------                                          --------------  -----  
 0   Country                                         500 non-null    str    
 1   Year                                            500 non-null    int64  
 2   Total Water Consumption (Billion Cubic Meters)  500 non-null    float64
 3   Per Capita Water Use (Liters per Day)           500 non-null    float64
 4   Agricultural Water Use (%)                      500 non-null    float64
 5   Industrial Water Use (%)                        500 non-null    float64
 6   Household Water Use (%)                         500 non-null    float64
 7   Rainfall Impact (Annual Precipitation in mm)    500 non-null    float64
 8   Groundwater Depletion Rate (%)                  500 non-null    float64
 9   Water Scarcity Level            

### Rangkuman Kualitas Data:
Tidak ditemukan data kosong atau baris duplikat, tipe data untuk semua kolom sudah sesuai, rentang nilai min max sudah masuk akal.
Jadi tidak ada data yang harus diganti atau dihapus.


In [10]:
scarcity_counts = df['Water Scarcity Level'].value_counts()
scarcity_pct = df['Water Scarcity Level'].value_counts(normalize=True) * 100
tabel_kategorik = pd.DataFrame({
    'Jumlah Negara': scarcity_counts,
    'Persentase (%)': scarcity_pct
})
display(tabel_kategorik)

,Jumlah Negara,Persentase (%)
Water Scarcity Level,,
Moderate,360,72.0
Low,122,24.4
High,18,3.6


### Interpretasi Variabel Kategorik (`Water Scarcity Level`)

Berdasarkan tabel distribusi frekuensi:
- Kategori Moderate mendominasi data secara signifikan, yaitu sebanyak 72,0%.
- Kategori Low berada di posisi kedua sebesar 24,4%.
- Kategori High memiliki proporsi paling sedikit, yaitu hanya 3,6%.

Distribusi ini menunjukkan bahwa sebagian besar negara dalam dataset berada pada tingkat kelangkaan air moderate atau sedang.

In [11]:
from IPython.display import display

kolom_numerik = [
    'Total Water Consumption (Billion Cubic Meters)',
    'Per Capita Water Use (Liters per Day)',
    'Rainfall Impact (Annual Precipitation in mm)'
]
ringkasan = []
for col in kolom_numerik:
    data = df[col]
    q1 = data.quantile(0.25)
    q3 = data.quantile(0.75)
    
    ringkasan.append({
        'Variabel': col,
        'Rata-rata': data.mean(),
        'Median': data.median(),
        'Modus': data.mode()[0],
        'Min': data.min(),
        'Max': data.max(),
        'Jangkauan': data.max() - data.min(),
        'Varians': data.var(),
        'Simpangan Baku': data.std(),
        'Q1': q1,
        'Q3': q3,
        'IQR': q3 - q1
    })
tabel_ringkasan = pd.DataFrame(ringkasan).set_index('Variabel').T
display(tabel_ringkasan)
sampel = df['Per Capita Water Use (Liters per Day)'].head(10)
n = len(sampel)
rerata_sampel = sampel.mean()
kuadrat_selisih = [(x - rerata_sampel)**2 for x in sampel]
varians_manual = sum(kuadrat_selisih) / (n - 1)
std_manual = varians_manual ** 0.5

print("Hasil Perhitungan Manual (10 Baris Pertama): ")
print("Rata-rata Sampel :", rerata_sampel)
print("Varians Manual   :", varians_manual, "| Pandas:", sampel.var())
print("Std Dev Manual   :", std_manual, "| Pandas:", sampel.std())

Variabel,Total Water Consumption (Billion Cubic Meters),Per Capita Water Use (Liters per Day),Rainfall Impact (Annual Precipitation in mm)
Rata-rata,501.224430,276.004782,1544.824300
Median,502.197154,276.430556,1537.537724
Modus,129.636667,111.708333,700.230000
Min,129.636667,111.708333,700.230000
Max,798.418000,404.350000,2533.678000
Jangkauan,668.781333,292.641667,1833.448000
Varians,9231.162101,1820.694137,85723.980639
Simpangan Baku,96.078937,42.669593,292.786579
Q1,441.447385,250.225406,1353.734583
Q3,563.849594,300.221750,1746.402425


Hasil Perhitungan Manual (10 Baris Pertama): 
Rata-rata Sampel : 246.28697107
Varians Manual   : 4478.414959259256 | Pandas: 4478.414959259256
Std Dev Manual   : 66.92096053748224 | Pandas: 66.92096053748224


### Interpretasi Statistik Deskriptif
1. Total Water Consumption:
Rata rata dan median nya mirip, yang menunjukan sebaran data nya simetris atau tidak miring
Rata-rata: **501,22 miliar m³**, Median: **502,20 miliar m³**.

3. Per Capita Water Use
Rata rata dan median berdekatan, yang menunjukan distribusi simetris.
Rata-rata: **276,00 liter/hari**, Median: **276,43 liter/hari**.

5. Rainfall Impact
Rata rata curah hujan tahunan sebesar 1544.82 mm/tahun dengan median 1537.54 mm/tahun (sebaran simetris)
Rata-rata: **1544,82 mm/tahun**, Median: **1537,54 mm/tahun**.

Hasil perhitungan manual varians dan simpangan baku sampel pada 10 baris pertama kolom Per Capita Water Use dengan pembagi n - 1 = 9:
- **Rata-rata Sampel:** `246.2870`
- **Varians Manual:** `4478.4150` (Cocok 100% dengan `.var()` Pandas)
- **Std Dev Manual:** `66.9210` (Cocok 100% dengan `.std()` Pandas)